pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleSchemaTest.java

```diff
@@ -131,9 +132,7 @@ public void newProducerWithoutSchemaOnTopicWithSchema() throws Exception {
         }
 
         try (Producer<byte[]> p = pulsarClient.newProducer().topic(topic).create()) {
-            Assert.fail("Shouldn't be able to connect to a schema'd topic with no schema");
-        } catch (PulsarClientException e) {
-            Assert.assertTrue(e.getMessage().contains("IncompatibleSchemaException"));
+            p.send("junkdata".getBytes(UTF_8));
```

Comment:
We also need the complimentary test, where schema _is_ being enforced (i.e. the original test).

Reply:
updated with the tests